### PR TITLE
Adding two new dimensions in metrics-bridge: hostname and agentRole

### PR DESCRIPTION
**Pull request by Makdaam**
both values are derived from the IP address in "instance".

```release-note
NONE
```

### CODE REVIEW

File: cmd/metricsbridge/metricsbridge.go [hunk 1]
```diff
@@ -233,6 +241,15 @@ func (c *config) runOnce(ctx context.Context) error {
 			for k, v := range sample.Metric {
 				if k != model.MetricNameLabel {
 					f.Dims[string(k)] = string(v)
+					//if there's an instance dimension try to lookup the hostname
+					if string(k) == "instance" {
```
Comment (jim-minter) [inline cmd/metricsbridge/metricsbridge.go]: nit: to my eye it'd be neater and clearer to move this out of the for loop and do `if instance, found := sample.Metric["instance"]; found {`

File: cmd/metricsbridge/metricsbridge.go [hunk 2]
```diff
@@ -233,6 +241,15 @@ func (c *config) runOnce(ctx context.Context) error {
...
+						hostname, present := hostnameMap[strings.Split(string(v), ":")[0]]
+						if present {
+							//only add the hostname dimension if the lookup succeeds
+							f.Dims["hostname"] = hostname
+							f.Dims["agentRole"] = strings.Split(hostname, "-")[0]
```
Comment (jim-minter) [inline cmd/metricsbridge/metricsbridge.go]: this doesn't work - users can choose their compute agent pool name so you could have a hostname (e.g.) `my-compute-1234567890-000000`.  You could use the following function: if the first part of the hostname is master -> master; if it's infra -> infra; anything else -> compute

Comment (jim-minter) [reply cmd/metricsbridge/metricsbridge.go]: s/agentRole/agentrole/ ?

File: cmd/metricsbridge/metricsbridge.go (hunk 1 above)
Comment (Makdaam) [reply cmd/metricsbridge/metricsbridge.go]: moved below, among other additional dimensions

File: cmd/metricsbridge/metricsbridge.go (hunk 2 above)
Comment (Makdaam) [reply cmd/metricsbridge/metricsbridge.go]: Changed capitalisation to "agentrole".

Changed role determination based on: "master","infra", everything else.

File: cmd/metricsbridge/metricsbridge.go [hunk 3]
```diff
@@ -235,6 +243,20 @@ func (c *config) runOnce(ctx context.Context) error {
 					f.Dims[string(k)] = string(v)
 				}
 			}
+			//if there's an instance dimension try to lookup the hostname
+			if instance, found := sample.Metric["instance"]; found {
+				hostname, present := hostnameMap[strings.Split(string(instance), ":")[0]]
+				if present {
+					//only add the hostname dimension if the lookup succeeds
+					f.Dims["hostname"] = hostname
+					hostnameprefix := strings.Split(hostname, "-")[0]
```
Comment (jim-minter) [inline cmd/metricsbridge/metricsbridge.go]: belated request: please could you put the function that calculates the agent pool profile role from the hostname into pkg/config/names.go with a unit test? that way we keep all our expectations about how these things map together into one place?

Comment (Makdaam) [reply cmd/metricsbridge/metricsbridge.go]: Moved function to GetAgentRole in pkg/config/names.go, added unit tests.

File: cmd/metricsbridge/metricsbridge.go [hunk 4]
```diff
@@ -22,6 +22,7 @@ import (
 	"github.com/prometheus/common/model"
 	"github.com/sirupsen/logrus"
 
+	pkgconfig "github.com/openshift/openshift-azure/pkg/config"
```
Comment (jim-minter) [inline cmd/metricsbridge/metricsbridge.go]: I preferred to rename the import than the struct in this case

File: pkg/config/names.go [hunk 5]
```diff
@@ -53,3 +54,16 @@ func GetScaleSetNameAndInstanceID(hostname string) (string, string, error) {
 
 	return "ss-" + hostname[:i], fmt.Sprintf("%d", instanceID), nil
 }
+
+// GetAgentRole parses a hostname, e.g. master-000000 or infra-12345-000000
+// and returns the lowercase role name ("master", "infra" or "compute")
+func GetAgentRole(hostname string) api.AgentPoolProfileRole {
```
Comment (jim-minter) [inline pkg/config/names.go]: preferred to return the enum type rather than a string

**Comment by jim-minter** [comment]
@Makdaam I pushed a couple of nitty fixes

/lgtm

**Pull request merged**